fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java

```diff
@@ -1148,12 +1146,17 @@ private boolean onReloadImpl(boolean postLoadImage) {
                 continue;
             } else if (table.isMaterializedView()) {
                 MaterializedView baseMV = (MaterializedView) table;
-                // skip reloading only when postLoadImage is true
-                if (postLoadImage && baseMV.isReloaded()) {
-                    continue;
+                // only consider skipping reload when postLoadImage is true
+                if (Config.enable_mv_reload_check_post_image && postLoadImage) {
+                    if (!baseMV.isReloaded()) {
+                        // recursive reload MV, to guarantee the order of hierarchical MV
```

Comment:
1. if `baseMV.isReloaded` is true, add a `logInfo` to trace it?
2. `hasReloaded` seems a better name?